app/apikey.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/temporalio/tcld/protogen/api/authservice/v1"
+	"github.com/urfave/cli/v2"
+	"google.golang.org/grpc"
+)
+
+type APIKeyClient struct {
+	authClient authservice.AuthServiceClient
+	ctx        context.Context
+}
+
+func NewAPIKeyClient(ctx context.Context, conn *grpc.ClientConn) *APIKeyClient {
+	return &APIKeyClient{
+		authClient: authservice.NewAuthServiceClient(conn),
+		ctx:        ctx,
+	}
+}
+
+type GetAPIKeyClientFn func(ctx *cli.Context) (*APIKeyClient, error)
+
+func GetAPIKeyClient(ctx *cli.Context) (*APIKeyClient, error) {
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return NewAPIKeyClient(ct, conn), nil
+}
+
+func (s *APIKeyClient) createAPIKey(
+	displayName string,
+	description string,
+	expiry time.Time,
+	requestID string,
+) error {
+	expiryts, err := types.TimestampProto(expiry)
+	if err != nil {
+		return fmt.Errorf("failed to convert timestamp to proto: %s", err)
+	}
+	resp, err := s.authClient.CreateAPIKey(s.ctx, &authservice.CreateAPIKeyRequest{
+		Spec: &auth.APIKeySpec{
+			DisplayName: displayName,
+			Description: description,
+			ExpiryTime:  expiryts,
+		},
+		RequestId: requestID,
+	})
+	if err != nil {
+		return err
+	}
+	return PrintProto(resp)
+}
+
+func (s *APIKeyClient) listAPIKey() error {
+
+	totalRes := &authservice.GetAPIKeysResponse{}
+	pageToken := ""
+	for {
+		resp, err := s.authClient.GetAPIKeys(s.ctx, &authservice.GetAPIKeysRequest{
+			PageToken: pageToken,
+		})
+		if err != nil {
+			return err
+		}
+		totalRes.ApiKeys = append(totalRes.ApiKeys, resp.ApiKeys...)
+		// Check if we should continue paging
+		pageToken = resp.NextPageToken
+		if len(pageToken) == 0 {
+			return PrintProto(totalRes)
+		}
+	}
+}
+
+func (s *APIKeyClient) getAPIKey(
+	id string,
+) (*auth.APIKey, error) {
+	resp, err := s.authClient.GetAPIKey(s.ctx, &authservice.GetAPIKeyRequest{
+		Id: id,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.ApiKey, nil
+}
+
+func (s *APIKeyClient) deleteAPIKey(
+	id string,
+	resourceVersion string,
+	requestID string,
+) error {
+	resp, err := s.authClient.DeleteAPIKey(s.ctx, &authservice.DeleteAPIKeyRequest{
+		Id:              id,
+		ResourceVersion: resourceVersion,
+		RequestId:       requestID,
+	})
+	if err != nil {
+		return err
+	}
+	return PrintProto(resp)
+}
+
+func NewAPIKeyCommand(getAPIKeyClientFn GetAPIKeyClientFn) (CommandOut, error) {
+	var c *APIKeyClient
+	return CommandOut{
+		Command: &cli.Command{
+			Name:    "apikey",
+			Aliases: []string{"ak"},
+			Usage:   "APIKey operations",
+			Before: func(ctx *cli.Context) error {
+				var err error
+				c, err = getAPIKeyClientFn(ctx)
+				return err
+			},
+			Subcommands: []*cli.Command{
+				{
+					Name:    "create",
+					Usage:   "Create an user apikey",
+					Aliases: []string{"c"},
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "name",
+							Usage:    "the display name of the apikey",
```

Comment:
nit: I wonder if we should replace all these instances of `apikey` with `API key` or some other form and keep it consistent in the other help messages we have in the CLI.